sources/packages/backend/apps/api/src/services/application/application.service.ts

```diff
@@ -1543,6 +1544,90 @@ export class ApplicationService extends RecordDataModelService<Application> {
     });
   }
 
+  /**
+   * Gets all eligible application that can be requested for application
+   * offering change.
+   * @param locationId location id.
+   * @param paginationOptions options to execute the pagination.
+   * @returns list of eligible application that can be requested for
+   * application offering change.
+   */
+  async getEligibleApplicationOfferingChangeRecords(
+    locationId: number,
+    paginationOptions: PaginationOptions,
+  ): Promise<PaginatedResults<Application>> {
+    const applicationQuery = this.repo
+      .createQueryBuilder("application")
+      .select([
+        "application.applicationNumber",
+        "application.id",
+        "currentAssessment.id",
+        "offering.studyStartDate",
+        "offering.studyEndDate",
+        "student.id",
+        "user.firstName",
+        "user.lastName",
+      ])
+      .leftJoin("application.currentAssessment", "currentAssessment")
+      .leftJoin("currentAssessment.offering", "offering")
+      .leftJoin(
+        "application.applicationOfferingChangeRequest",
+        "applicationOfferingChangeRequest",
+      )
+      .innerJoin("application.student", "student")
+      .innerJoin("student.user", "user")
+      .where(
+        new Brackets((qb) =>
+          qb
+            .where("applicationOfferingChangeRequest.id IS NULL")
+            .orWhere(
+              "applicationOfferingChangeRequest.applicationOfferingChangeRequestStatus NOT IN (:...status)",
+              {
+                status: [
+                  ApplicationOfferingChangeRequestStatus.InProgressWithSABC,
+                  ApplicationOfferingChangeRequestStatus.InProgressWithStudent,
```

Comment:
Should not include "not in `DeclinedByStudent` and `DeclinedBySABC`"?

Reply:
in that case we need to include Approved as well. Above condition is relatively better IMO.

Reply:
I think Declined requests become re-eligible for offering change. @ann-aot can confirm.

Reply:
no @andrepestana-aot, if a request is in any completed state (declined/approved) then institution will be again able to request another change if it's still not achieved